src/app/admin/user-panel/user-panel.component.ts

```diff
@@ -34,10 +54,42 @@ export class AdminPanelComponent implements OnInit {
   private webApiService = inject(WebApiService);
   authService = inject(AuthService);
 
+  // User management table data
   tableUsersList = new MatTableDataSource<User>([]);
   displayedColumns = ['email', 'roles', 'actions'];
   selectedUser: User | undefined = undefined;
 
+  // Tab state
+  selectedTabIndex = 0;
+
+  // Displayed columns for the user logs
+  userLogsDisplayedColumns = ['email', 'action', 'time'];
+
+  // Pagination state
+  private pageSubject = new BehaviorSubject({
+    page: 1,
+    limit: 50,
+  });
+
+  userLogsLoading = false;
+  userLogsError: string | null = null;
+  userLogs$ = this.pageSubject.pipe(
```

Comment:
yes, this needs to be using signals, but once you make the change, it exposes another problem
```typescript
  userLogs$ = this.pageSubject.pipe(
    switchMap(({ page, limit }) => {
      console.log('loading true');
      this.userLogsLoading.set(true);
      return this.webApiService.userLogs({ page, limit }).pipe(
        catchError((err: any) => {
          this.userLogsError = err as string;
          return of(null);
        }),
        finalize(() => {
          console.log('loading false');
          this.userLogsLoading.set(false);
        })
      );
    })
  );
```

the finalize logic makes sense, but the problem is in the template you have:
```
          @if (userLogsLoading()) {
            <mat-spinner />
          } @else if (userLogsError) {
            <p>An error occurred... {{ userLogsError }}</p>
          } @else {
            <ng-container *ngIf="userLogs$ | async as logs">
```

So whenever it gets set to false, it the AsyncPipe resubscribes, then the code in `switchMap` runs again. 

One solution is to just always display the table.